float32 x
float32 y
float32 theta
string img_name
bool with_collision
string name # Optional.  A unique name will be created and returned if this is empty
---
string name